lib/vector-core/buffers/src/test/mod.rs

```diff
@@ -0,0 +1,54 @@
+mod common;
+mod model;
+
+use crate::{Acker, DropWhenFull};
+use futures::task::Poll;
+use futures::{channel::mpsc, future, task::AtomicWaker};
+use futures::{Sink, Stream};
+use std::sync::{atomic::AtomicUsize, Arc};
+use tokio_test::task::spawn;
+
+#[tokio::test]
+async fn drop_when_full() {
+    future::lazy(|cx| {
+        let (tx, rx) = mpsc::channel(2);
+
+        let mut tx = Box::pin(DropWhenFull::new(tx));
+
+        assert_eq!(tx.as_mut().poll_ready(cx), Poll::Ready(Ok(())));
+        assert_eq!(tx.as_mut().start_send(1), Ok(()));
+        assert_eq!(tx.as_mut().poll_ready(cx), Poll::Ready(Ok(())));
+        assert_eq!(tx.as_mut().start_send(2), Ok(()));
+        assert_eq!(tx.as_mut().poll_ready(cx), Poll::Ready(Ok(())));
+        assert_eq!(tx.as_mut().start_send(3), Ok(()));
+        assert_eq!(tx.as_mut().poll_ready(cx), Poll::Ready(Ok(())));
+        assert_eq!(tx.as_mut().start_send(4), Ok(()));
+
+        let mut rx = Box::pin(rx);
+
+        assert_eq!(rx.as_mut().poll_next(cx), Poll::Ready(Some(1)));
+        assert_eq!(rx.as_mut().poll_next(cx), Poll::Ready(Some(2)));
+        assert_eq!(rx.as_mut().poll_next(cx), Poll::Ready(Some(3)));
+        assert_eq!(rx.as_mut().poll_next(cx), Poll::Pending);
```

Comment:
I'm not seeing why this test passes. It appears to send four values but show that only three are ready to receive.

Reply:
You know, that's a good question. I didn't write this test for the PR, just moved it, so I didn't look at it too closely. Worth a follow up ticket I think.